packages/builder/src/components/backend/DataTable/DataTable.svelte

```diff
@@ -27,6 +27,17 @@
   $: isUsersTable = id === TableNames.USERS
   $: isInternal = $tables.selected?.type !== "external"
 
+  $: datasource = $datasources.list.find(datasource => {
+    return datasource._id === $tables.selected.sourceId
```

Comment:
Would maybe just quickly double check that if you deleted the selected table that this statement doesn't fire with `$tables.selected` being undefined, since that would crash. Could also just add in optional chaining to be extra safe!